certbot_dns_desec/dns_desec.py

```diff
@@ -4,17 +4,31 @@
 import time
 
 import requests
-import zope.interface
-from certbot import errors
 from certbot import interfaces
+try:
+    # needed for compatibility with older certbots, see #13
+    import zope.interface
+    zope_interface_implementer = zope.interface.implementer
+    zope_interface_provider = zope.interface.provider
+    i_authenticator = interfaces.IAuthenticator
+    i_plugin_factory = interfaces.IPluginFactory
+except ImportError:
+    def get_noop_dec(*args):
+        def noop_dec(obj):
+            return obj
+        return noop_dec
```

Comment:
Shouldn't this be (observe argument placement and `obj()` call)
```
python
def get_noop_dec(obj):
    def noop_dec(*args, **kwargs):
        obj(*args, **kwargs)
    return noop_dec
```
or even
```python
def get_noop_dec(obj):
    return obj
```

When making a change anyway, you could also rename `dec` to `decorator` (looks like `decode` otherwise).

Reply:
In 

```python3
@dec(1, 2)
def foo():
    pass
```

the decorator is the **return value** of `dec(1, 2)`, this is hence equivalent to

```python3
my_dec = dec(1,2)

@my_dec
def foo():
    pass
```

This, in turn, is equivalent to replacing `foo(...)` by `my_dec(f)(...)`.

Hence, in our code, in order to not modify the functions behavior, `noop_dec` needs to be the identity function. As the decorator is called with one argument, `noop_dec` needs to be the return value of a function that ignores the first argument (here named `get_noop_dec`).

Does that answer your question?

Reply:
I named the first argument to `noop_dec` `obj` because the decorator is applied to a class, not a function. (Using `f` or `func` would be confusing, `cls` is usually used in different context, `class` cannot be used.)

Reply:
> Does that answer your question?

Yes, thanks. I was not aware of the different treatment for decorators with arguments.